test/dynamo/test_minifier.py

```diff
@@ -47,65 +151,445 @@ def setUp(self):
+            # Make sure to comment out `shutil.rmtree...` above as well.
+            print("repro dir:", repro_dir_match.group(1))
+            return proc, repro_dir_match.group(1)
+        return proc, None
+
+    # Patch generated files with testing patches
+    def _inject_code(self, patch_code, filename):
+        with open(filename, "r") as f:
+            code = f.read()
+        code = code.replace(TEST_REPLACEABLE_COMMENT, patch_code)
+        with open(filename, "w") as f:
+            f.write(code)
+        return code
+
+    # Runs the minifier launcher script in `repro_dir`, patched with `patch_code`.
+    def _run_minifier_launcher(self, patch_code, repro_dir):
+        self.assertIsNotNone(repro_dir)
+        launch_file = os.path.join(repro_dir, "minifier_launcher.py")
+        self.assertTrue(os.path.exists(launch_file))
+        launch_code = self._inject_code(patch_code, launch_file)
+        launch_proc = subprocess.run(
+            ["python3", launch_file],
+            capture_output=True,
+            cwd=repro_dir,
+        )
+        return launch_proc, launch_code
+
+    # Runs the repro script in `repro_dir`, patched with `patch_code`
+    def _run_repro(self, patch_code, repro_dir):
+        self.assertIsNotNone(repro_dir)
+        repro_file = os.path.join(repro_dir, "repro.py")
         self.assertTrue(os.path.exists(repro_file))
+        repro_code = self._inject_code(patch_code, repro_file)
 
-    # If error_at_aot is True, an error will be produced when AOTAutograd
-    # attempts to generate the backward graph.
-    # If error_after_aot is False, an error will be produced in inductor.
-    def _test_around_aot(self, error_at_aot):
-        mod = MockModule()
-        opt_mod = torch._dynamo.optimize("inductor")(mod)
+        repro_proc = subprocess.run(["python3", repro_file], capture_output=True)
 
-        repro_file = torch._dynamo.debug_utils.get_minifier_repro_path()
-        repro_after = "dynamo" if error_at_aot else "aot"
+        return repro_proc, repro_code
 
-        @patch.object(torch._dynamo.config, "repro_after", repro_after)
-        def inner():
-            x = torch.randn(4)
-            x.requires_grad = error_at_aot
-            try:
-                opt_mod(x)
-            except Exception:
-                pass
+    # Template for testing code.
+    # `run_code` is the code to run for the test case.
+    # `patch_code` is the code to be patched in every generated file.
+    def _gen_test_code(self, run_code, repro_after, repro_level, patch_code):
+        return f"""\
+import torch
+import torch._dynamo
+{patch_code}
+torch._dynamo.config.repro_after = "{repro_after}"
+torch._dynamo.config.repro_level = {repro_level}
 
-        inner()
+{run_code}
+"""
 
-        self.assertTrue(os.path.exists(repro_file))
+    # Runs a full minifier test.
+    # Minifier tests generally consist of 3 stages:
+    # 1. Run the problematic code (in a separate process since it could segfault)
```

Comment:
Did you add tests for the segfault case? Nice!

Reply:
The after aot C++ tests abort (similar to segfault).